include/seqan3/core/detail/debug_stream_named_enumeration.hpp

```diff
@@ -0,0 +1,52 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief Provides an overload for the seqan3::debug_stream if a seqan3::enumeration_names of the type exists.
+ */
+
+#pragma once
+
+#include <variant>
+
+#include <seqan3/argument_parser/auxiliary.hpp>
+#include <seqan3/core/detail/debug_stream_type.hpp>
+
+namespace seqan3
+{
+/*!\name Formatted output overloads
+ * \{
+ */
+/*!\brief A type (e.g. an enum) can be made debug streamable by customizing the seqan3::enumeration_names.
+ * \tparam option_type Type of the enum to be printed.
+ * \param s  The seqan3::debug_stream.
+ * \param op The value to print.
+ * \relates seqan3::debug_stream_type
+ *
+ * \details
+ *
+ * This searches the seqan3::enumeration_names of the respective type for the value \p op and prints the
+ * respective string if found or '\<UNKNOWN_VALUE\>' if the value cannot be found in the map.
+ */
+template <typename char_t, typename option_type>
+//!\cond
+    requires named_enumeration<remove_cvref_t<option_type>>
+//!\endcond
+inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, option_type && op)
+{
+    for (auto & [key, value] : enumeration_names<option_type>)
+    {
+        if (op == value)
+            return s << key;
+    }
+
+    return s << "<UNKNOWN_VALUE>";
+}
+//!\}
+
+} // namespace seqan3
```

Comment:
I would move this to the file that defines the concept! The headers in core are reserved for types/concepts from std.